### PR TITLE
Calling samples page fix

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES # https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-428150 

## This pull request addresses

Test calling samples page, check all the features and fix if something is not working.

## by making the following changes

Manually moved contacts changes that were merged till 1.68.0 version of web-calling-sdk to calling folder inside js-sdk. That fixed contacts as well as voicemail/call settings

Below commit changes from web-calling-sdk repo has been merged now:

https://sqbu-github.cisco.com/WebExSquared/web-calling-sdk/commit/90ef4fd8e48239a8b4e7c5a4f04c3c47650175c9
https://sqbu-github.cisco.com/WebExSquared/web-calling-sdk/commit/54f6b72044710925fe843187e2dfc19a9ae3467b 
https://sqbu-github.cisco.com/WebExSquared/web-calling-sdk/commit/a88e0621b761ca82ce7175ab7aaaea3c5fd16da1 

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/calling/calling/src/Contacts/ContactsClient.ts [hunk 1]
```diff
@@ -52,160 +77,255 @@ export class ContactsClient implements IContacts {
...
+            );
+
+            decryptedAddressInfo = Object.fromEntries(
+              await Promise.all(decryptedAddressInfoPromises)
+            );
+          }
+
+          return [field, decryptedAddressInfo];
+        }
+        case 'emails':
```
Comment (arun3528) [inline packages/calling/calling/src/Contacts/ContactsClient.ts:231]: do you want to make these constants as key value

File: packages/calling/calling/src/Contacts/ContactsClient.ts [hunk 2]
```diff
@@ -52,160 +77,255 @@ export class ContactsClient implements IContacts {
...
+   * Decrypt emails, phoneNumbers, sipAddresses.
+   *
+   * @param contactDetails - Array ContactDetail.
+   * @param encryptionKeyUrl - KMS encryptionKeyUrl for the contact.
+   * @returns Decrypted phoneNumbers.
    */
-  private async decryptContact(
-    contact: Contact,
-    avatarUrlDomain: string,
+  private async decryptContactDetail(
```
Comment (arun3528) [inline packages/calling/calling/src/Contacts/ContactsClient.ts:86]: nice to have it on the global name space we could reuse in other places if needed

File: packages/calling/calling/src/Contacts/ContactsClient.ts [hunk 3]
```diff
@@ -52,160 +77,255 @@ export class ContactsClient implements IContacts {
...
+            );
+
+            encryptedAddressInfo = Object.fromEntries(
+              await Promise.all(encryptedAddressInfoPromises)
+            );
+          }
+
+          return [field, encryptedAddressInfo];
+        }
+        case 'emails':
```
Comment (arun3528) [inline packages/calling/calling/src/Contacts/ContactsClient.ts:162]: same here have it as constants in this file

File: packages/calling/calling/src/Contacts/constants.ts [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+  // 'avatarURL',
+  'companyName',
+  'displayName',
+  'emails',
+  'firstName',
+  'lastName',
+  'phoneNumbers',
+  'sipAddresses',
+  'title',
+];
```
Comment (BhargavSatya) [inline packages/calling/calling/src/Contacts/constants.ts:20]: you could make this as an enum and use this as key-value for what @arun3528 is suggesting. You can maybe convert it into array wherever needed as an array of encryptedFields.

**Review by BhargavSatya — APPROVED** [review]
A minor suggestion.

File: docs/samples/calling/index.html [hunk 5]
```diff
@@ -153,7 +153,7 @@ <h2>Call initialization</h2>
           </fieldset>
         </form>
 
-        <!-- meeting-controls / media-devices -->
+        <!-- meeting-controls / media-devices
```
Comment (sreenara) [inline docs/samples/calling/index.html:156]: Could we add a TODO or something here so we can bring it back when needed?

File: packages/webex/src/calling.js [hunk 6]
```diff
@@ -4,6 +4,7 @@ import EventEmitter from 'events';
 /* eslint-disable require-jsdoc */
 require('@webex/internal-plugin-device');
 require('@webex/internal-plugin-mercury');
+require('@webex/internal-plugin-encryption');
```
Comment (sreenara) [inline packages/webex/src/calling.js:7]: What did this change fix?

**Review by sreenara — CHANGES_REQUESTED** [review]
Please check the comments.
Update the PR description with the list of commit links which were added here so we can use them as reference for future.

File: packages/webex/src/calling.js (hunk 6 above)
Comment (Kesari3008) [reply packages/webex/src/calling.js:7]: this is needed to make encryption service work in contacts

**Review by arun3528 — APPROVED** [review]

**Review by sreenara — APPROVED** [review]

**Pull request merged**